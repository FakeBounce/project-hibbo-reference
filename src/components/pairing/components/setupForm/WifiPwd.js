import React from 'react';
import { Animated } from 'react-native';
import { shape, string, func, bool, instanceOf } from 'prop-types';

import appStyles from 'styles/appStyles';

import { getTranslations } from 'utils/i18n';

import { StyledContainerBasic } from 'styledComponents/containers';

import Title from '../Title';
import Input from './Input';

import * as Styles from './styles';

const WifiPwd = ({
  form,
  setData,
  validate,
  error,
  controls,
  title,
  titleAnimations,
  inputAnimations,
  controlsAnimations,
  autoFocus,
  showContent,
}) => {
  return (
    <StyledContainerBasic>
      <Title
        label={[title.label1, title.label2, title.label3, title.label4]}
        color={appStyles.colors.white}
        animations={titleAnimations}
        showContent={showContent}
      />

      <Styles.AnimatedWrapperPwd>
        {/* <Styles.SCenteredPwd>
          <Styles.STextPwd>{form.wifiName}</Styles.STextPwd>
        </Styles.SCenteredPwd> */}
        <Input
          name="wifiPwd"
          defaultValue={form.wifiPwd}
          placeholder={getTranslations('pairing.pwd')}
          setData={setData}
          validate={validate}
          error={error}
          secureEntry
          animations={inputAnimations}
          autoFocus={autoFocus}
        />
      </Styles.AnimatedWrapperPwd>

      <Styles.SControls
        rightLabel={controls.rightLabel}
        rightAction={() => {
          validate('wifiPwd');
        }}
        animations={controlsAnimations}
      />
    </StyledContainerBasic>
  );
};

WifiPwd.defaultProps = {
  showContent: null,
};

WifiPwd.propTypes = {
  title: shape({
    label1: string,
    label2: string,
    label3: string,
    label4: string,
  }).isRequired,
  form: shape({
    wifiName: string.isRequired,
    wifiPwd: string.isRequired,
  }).isRequired,
  setData: func.isRequired,
  validate: func.isRequired,
  error: bool.isRequired,
  controls: shape({
    rightLabel: string,
  }).isRequired,
  titleAnimations: shape({
    opacity: instanceOf(Animated.Value).isRequired,
    translateY: instanceOf(Animated.Value).isRequired,
    scaleY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  inputAnimations: shape({
    borderOpacity: instanceOf(Animated.Value).isRequired,
    borderTranslateY: instanceOf(Animated.Value).isRequired,
    inputOpacity: instanceOf(Animated.Value).isRequired,
    inputTranslateY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  controlsAnimations: shape({
    left: shape({
      opacity: instanceOf(Animated.Value).isRequired,
      translateY: instanceOf(Animated.Value).isRequired,
      scale: instanceOf(Animated.Value).isRequired,
    }).isRequired,
    right: shape({
      opacity: instanceOf(Animated.Value).isRequired,
      translateY: instanceOf(Animated.Value).isRequired,
      scale: instanceOf(Animated.Value).isRequired,
    }).isRequired,
  }).isRequired,
  autoFocus: bool.isRequired,
  showContent: func,
};

export default WifiPwd;
