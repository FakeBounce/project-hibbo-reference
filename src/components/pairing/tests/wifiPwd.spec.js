import React from 'react';
import renderer from 'react-test-renderer';

import WifiPwd from '../components/setupForm/WifiPwd';

describe('WifiPwd', () => {
  const form = {
    wifiName: '',
    wifiPwd: '',
  };
  const fn = jest.fn();
  const controls = {
    rightLabel: '',
  };
  const component = renderer.create(
    <WifiPwd
      form={form}
      setData={fn}
      validate={fn}
      error={false}
      controls={controls}
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
