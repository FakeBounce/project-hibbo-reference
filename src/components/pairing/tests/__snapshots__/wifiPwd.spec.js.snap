// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WifiPwd should render the component 1`] = `
<View
  style={undefined}
>
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {},
        ]
      }
    >
      <TextInput
        autoCapitalize="words"
        autoCorrect={false}
        autoFocus={false}
        keyboardType="default"
        maxLength={100}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Password"
        placeholderTextColor="#FFDF2D"
        returnKeyType="send"
        secureTextEntry={true}
        style={
          Array [
            undefined,
            Object {},
            Object {
              "color": "#FFDF2D",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    collapsable={undefined}
    style={Object {}}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        
      </Text>
    </View>
  </View>
</View>
`;
